File: 242-valid-anagram/valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        unordered_map<char,int> mp;
        for(auto it:s){
            mp[it]++;
        }
        for(auto it:t){
            mp[it]--;
        }
        for(auto x: mp){
            if(x.second!=0){
                return false;
            }
        }
        return true;
    }
};

File: 242-valid-anagram/valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        seen= {}
        if len(s) != len(t):
            return False
        for num in s:
            if num in seen:
                seen[num]+=1
            else:
                seen[num]=1

        for num in t:
            if num in seen:
                seen[num]-=1
    
        for num in seen:
            if seen[num] != 0:
                return False
        return True